fvm/fvm_bench_test.go

```diff
@@ -0,0 +1,972 @@
+			cadenceValues := make([]cadence.Value, testTokensPerTransaction)
+			startTestToken := (i*transactionsPerBlock+j)*testTokensPerTransaction + 1
+			for m := 0; m < testTokensPerTransaction; m++ {
+				cadenceValues[m] = cadence.NewUInt64(uint64(startTestToken + m))
+			}
+
+			encodedArg, err := jsoncdc.Encode(
+				cadence.NewArray(cadenceValues),
+			)
+			require.NoError(b, err)
+
+			txBody := flow.NewTransactionBody().
+				SetScript(transferTx).
+				SetProposalKey(tctx.chain.ServiceAddress(), 0, tctx.serviceAccountSequenceNumber()).
+				AddAuthorizer(accounts[1]).
+				AddArgument(encodedArg).
+				AddArgument(encodedAddress).
+				SetPayer(tctx.chain.ServiceAddress())
+
+			err = testutil.SignPayload(txBody, accounts[1], privateKeys[1])
+			require.NoError(b, err)
+
+			err = testutil.SignEnvelope(txBody, tctx.chain.ServiceAddress(), unittest.ServiceAccountPrivateKey)
+			require.NoError(b, err)
+
+			transactions[j] = txBody
+		}
+
+		computationResult = executeBlocks([][]*flow.TransactionBody{transactions})
+		for j := 0; j < transactionsPerBlock; j++ {
+			require.Empty(b, computationResult.TransactionResults[j].ErrorMessage)
+		}
+	}
+}
+
+func prepareExecutionEnv(tb testing.TB, chain flow.Chain) func(txs [][]*flow.TransactionBody) *execution.ComputationResult {
+	rt := fvm.NewInterpreterRuntime()
+	vm := fvm.NewVirtualMachine(rt)
+
+	logger := zerolog.Nop()
+
+	opts := []fvm.Option{
+		fvm.WithTransactionFeesEnabled(true),
+		fvm.WithAccountStorageLimit(true),
+		fvm.WithChain(chain),
+	}
+
+	fvmContext :=
+		fvm.NewContext(
+			logger,
+			opts...,
+		)
+
+	collector := metrics.NewNoopCollector()
+	tracer := trace.NewNoopTracer()
+
+	wal := &fixtures.NoopWAL{}
+
+	ledger, err := completeLedger.NewLedger(wal, 100, collector, logger, completeLedger.DefaultPathFinderVersion)
+	require.NoError(tb, err)
+
+	bootstrapper := bootstrapexec.NewBootstrapper(logger)
+
+	initialCommit, err := bootstrapper.BootstrapLedger(
+		ledger,
+		unittest.ServiceAccountPublicKey,
+		chain,
+		fvm.WithInitialTokenSupply(unittest.GenesisTokenSupply),
+		fvm.WithAccountCreationFee(fvm.DefaultAccountCreationFee),
+		fvm.WithMinimumStorageReservation(fvm.DefaultMinimumStorageReservation),
+		fvm.WithTransactionFee(fvm.DefaultTransactionFees),
+		fvm.WithStorageMBPerFLOW(fvm.DefaultStorageMBPerFLOW),
+	)
+
+	require.NoError(tb, err)
+
+	ledgerCommitter := committer.NewLedgerViewCommitter(ledger, tracer)
+
+	blockComputer, err := computer.NewBlockComputer(vm, fvmContext, collector, tracer, logger, ledgerCommitter)
+	require.NoError(tb, err)
+
+	view := delta.NewView(exeState.LedgerGetRegister(ledger, initialCommit))
+
+	return func(txs [][]*flow.TransactionBody) *execution.ComputationResult {
+		executableBlock := unittest.ExecutableBlockFromTransactions(txs)
+		executableBlock.StartState = &initialCommit
+
+		computationResult, err := blockComputer.ExecuteBlock(context.Background(), executableBlock, view, programs.NewEmptyPrograms())
```

Comment:
it might be better for some transactions if we share the same programs.

Reply:
like for sequence of batch transfer transactions.

Reply:
I wanted to replicate actual behaviour, and I think we create new programs for each block, but now I'm not sure. I will check.

Reply:
the actual behaviour is keeping the programs between blocks until a new contract is deployed or updated

Reply:
Using a single instance of programs should take care of that, since programs are reset in the FVM